runelite-client/src/main/java/net/runelite/client/plugins/bankvalue/BankValuePlugin.java

```diff
@@ -0,0 +1,381 @@
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.bankvalue;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.inject.Inject;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.ItemComposition;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.MenuOptionClicked;
+import net.runelite.api.events.WidgetHiddenChanged;
+import net.runelite.api.queries.BankItemQuery;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetID;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.api.widgets.WidgetItem;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.util.StackFormatter;
+import net.runelite.http.api.item.ItemPrice;
+import static net.runelite.api.ItemID.COINS;
+
+@PluginDescriptor(name = "Bank Value")
+@Slf4j
+public class BankValuePlugin extends Plugin
+{
+	private static final float HIGH_ALCHEMY_CONSTANT = 0.6f;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private BankValueConfig config;
+
+	@Inject
+	private ItemManager itemManager;
+
+	private Thread bankThread = null;
+	private ExecutorService executor = null;
+
+	// Only meant to be used to tell if the player is in the bank (or during shutdown)
+	private boolean bankActive = false;
+
+	private String bankTitle = "The Bank of Runescape";
```

Comment:
This should be constant and not mutable variable, and remove the mutations of it.

Reply:
It's intentional. This variable gets set before the bank value gets put into the bank title. When changing the configs, the bank title gets reset by using that variable. This also goes for disabling the plugin while using the bank.

Reply:
I think deathbeam's comment applies here whether or not the mutability was intentional on your part. "The Bank of Runescape" is a string constant, so storing it in a mutable variable like this seems questionable at best

Reply:
Okay, I see the point. It was partly done in case the bank title got called to be reset before the variable itself got updated. But in that case, the bank would already have the correct title, and would not need to be reset. I'll update that.